src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Raw.cs

```diff
@@ -0,0 +1,107 @@
+        /// <param name="skipInputValidation">Whether to skip validation of the input JSON content.</param>
+        public void WriteRawValue(string json, bool skipInputValidation = false)
+        {
+            if (json == null)
+            {
+                throw new ArgumentNullException(nameof(json));
+            }
+
+            WriteRawValue(json.AsSpan(), skipInputValidation);
+        }
+
+        /// <summary>
+        /// Writes the input as JSON content.
+        /// </summary>
+        /// <param name="json">The raw JSON content to write.</param>
+        /// <param name="skipInputValidation">Whether to skip validation of the input JSON content.</param>
+        public void WriteRawValue(ReadOnlySpan<char> json, bool skipInputValidation = false)
+        {
+            byte[]? tempArray = null;
+
+            // For performance, avoid obtaining actual byte count unless memory usage is higher than the threshold.
+            Span<byte> utf8Json = json.Length <= (JsonConstants.ArrayPoolMaxSizeBeforeUsingNormalAlloc / JsonConstants.MaxExpansionFactorWhileTranscoding) ?
+                // Use a pooled alloc.
+                tempArray = ArrayPool<byte>.Shared.Rent(json.Length * JsonConstants.MaxExpansionFactorWhileTranscoding) :
+                // Use a normal alloc since the pool would create a normal alloc anyway based on the threshold (per current implementation)
+                // and by using a normal alloc we can avoid the Clear().
+                new byte[JsonReaderHelper.GetUtf8ByteCount(json)];
+
+            try
+            {
+                int actualByteCount = JsonReaderHelper.GetUtf8FromText(json, utf8Json);
+                utf8Json = utf8Json.Slice(0, actualByteCount);
+                WriteRawValue(utf8Json, skipInputValidation);
+            }
+            finally
+            {
+                if (tempArray != null)
+                {
+                    utf8Json.Clear();
+                    ArrayPool<byte>.Shared.Return(tempArray);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Writes the input as JSON content.
+        /// </summary>
+        /// <param name="utf8Json">The raw JSON content to write.</param>
+        /// <param name="skipInputValidation">Whether to skip validation of the input JSON content.</param>
+        public void WriteRawValue(ReadOnlySpan<byte> utf8Json, bool skipInputValidation = false)
+        {
+            if (utf8Json.Length == 0)
+            {
+                ThrowHelper.ThrowArgumentException(SR.ExpectedJsonTokens);
+            }
+
+            if (!skipInputValidation)
+            {
+                Utf8JsonReader reader = new Utf8JsonReader(utf8Json);
+
+                try
+                {
+                    while (reader.Read());
+                }
+                catch (JsonReaderException ex)
+                {
+                    ThrowHelper.ThrowArgumentException(ex.Message);
+                }
```

Comment:
Hmm, interesting. I wonder if we should just let the `JsonReaderException` bubble up, similar to `JsonDocument.Parse`, rather than capturing and throwing some other exception instead.

Reply:
We could do that, but given we already have a precedence of throwing `JsonReaderException` directly from other shipped APIs, the simplest approach works well here too. We don't throw `ArgumentException` in `JsonDocument.Parse` for invalid json input:
```C#
var reader = new Utf8JsonReader(utf8Json);
while (reader.Read());
```

So, if we had to pick one, internal consistency within STJ is kinda nice (+ less code :))